zip-0244.rst

```diff
@@ -435,37 +549,32 @@ If the sighash type is ``SIGHASH_SINGLE`` and the signature hash is being comput
 the transparent input at a particular index, and a transparent output appears in
 the transaction at that index::
 
-   the hash is updated with the transaction serialized form of the
-   transparent output at that index, and finalized.
+   the hash is over the transaction serialized form of the transparent output at that
+   index
 
-If the sighash type is ``SIGHASH_SINGLE`` and the signature is being computed for
-a shielded input, or if the sighash type is ``SIGHASH_NONE``::
+otherwise::
 
-   the hash is immediately finalized, without being updated with any
-   additional data
+   BLAKE2b-256(``ZTxIdOutputsHash``, [])
 
 S.2d: txin_sig_digest
 '''''''''''''''''''''
-This is a BLAKE2b-256 hash initialized with the personalization field value
-``Zcash___TxInHash`` (3 underscores).
-
-If the signature hash is being computed for a transparent input, the hash
-is updated with the following properties of that input::
+This is a BLAKE2b-256 hash of the following properties of the transparent input being
+signed, initialized with the personalization field value ``Zcash___TxInHash`` (3
+underscores).
 
    S.2d.i:   prevout     (field encoding)
    S.2d.ii:  script_code (field encoding)
    S.2d.iii: value       (8-byte signed little-endian)
```

Comment:
Hmm, I just spotted this. The value here should be unsigned (as we cannot have coins with negative values).

In practice it doesn't make much difference, as the upper range of a coin value would get coerced to the same encoding as signed integers, but is completely out of range for MAX_MONEY so won't ever be encountered. And I presume this snuck in because the `CAmount` type in `zcashd` is an `int64_t` (because it is dual-purposed both for non-negative and negative-allowed uses).

Reply:
Zebra also uses `i64` internally to convert to bytes:
https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/amount.rs#L49

But our `NonNegative` constraint ensures that the value is positive and below `MAX_MONEY`:
https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/amount.rs#L336

Reply:
If it's required to be nonnegative then it should be described as unsigned.

Reply:
@nuttycom and @dconnolly and I reviewed this and think it's fine as-is.